pos_fixed_discount/static/src/js/discount.js

```diff
@@ -0,0 +1,89 @@
+odoo.define("pos_fixed_discount.FixedDiscountButton", function (require) {
+    "use strict";
+
+    const PosComponent = require("point_of_sale.PosComponent");
+    const ProductScreen = require("point_of_sale.ProductScreen");
+    const {useListener} = require("web.custom_hooks");
+    const Registries = require("point_of_sale.Registries");
+
+    class FixedDiscountButton extends PosComponent {
+        constructor() {
+            super(...arguments);
+            useListener("click", this.onClick);
+        }
+        async onClick() {
+            var self = this;
+            const {confirmed, payload} = await this.showPopup("NumberPopup", {
+                title: this.env._t("Discount Amount"),
+                startingValue: 0,
+            });
+            if (confirmed) {
+                var val = Math.round(Math.max(0, Math.min(100, parseFloat(payload))));
+                self.apply_discount(val);
+            }
+        }
+
+        async apply_discount(amount) {
+            var order = this.env.pos.get_order();
+            var lines = order.get_orderlines();
+            var product = this.env.pos.db.get_product_by_id(
+                this.env.pos.config.discount_product_id[0]
+            );
+            if (product === undefined) {
+                await this.showPopup("ErrorPopup", {
+                    title: this.env._t("No discount product found"),
+                    body: this.env._t(
+                        "The discount product seems misconfigured. Make sure it is flagged as 'Can be Sold' and 'Available in Point of Sale'."
+                    ),
+                });
+                return;
+            }
+
+            // Remove existing discounts
+            var i = 0;
+            while (i < lines.length) {
+                if (lines[i].get_product() === product) {
+                    order.remove_orderline(lines[i]);
+                } else {
+                    i++;
+                }
+            }
```

Comment:
```suggestion
            // Remove existing discounts
            for (const line of lines) {
                if (line.get_product() === product) {
                    order.remove_orderline(line);
                }
            }
```

Doesn't this do the same?